sdk/core/Azure.Core/src/ChainingClassifier.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Core
+{
+    internal class ChainingClassifier : ResponseClassifier
+    {
+        private ResponseClassificationHandler[]? _handlers;
+        private ResponseClassifier _endOfChain;
+
+        public ChainingClassifier((int Status, bool IsError)[]? statusCodes,
+            ResponseClassificationHandler[]? handlers,
+            ResponseClassifier endOfChain)
```

Comment:
Would it make sense to eliminate the `endOfChain` parameter and let the ordering of the handlers indicate that concept? Since the ordering of the handlers is already a concern for callers, this seems like it would simplify the implementation.

Reply:
The chain is shaped like this:

- handler
- handler
- response classifier

So, while the handlers are ordered (and we state in public docstrings what that ordering is), there is still the base default classifier that ends the chain, and we need to have that passed-in for it to take effect.

This implements the same logic as `CoreResponseClassifier`, it's just less efficient because it holds a separate handler for status codes, whereas in `CoreResponseClassifier` the "end of chain" classifier has the status code classifications "smooshed" into it already.

Reply:
Ah, I had missed that endOfChain was a different type. For my suggestion to be valid, ResponseClassifier would need to be an instance of ResponseClassifierHandler that was guaranteed to have a classification.

Reply:
Makes sense!  The heterogenous chain and where in the chain different logic lands in the actual implementation is part of what's tricky about this area.